watertap/property_models/tests/test_seawater_ion_generic.py

```diff
@@ -0,0 +1,79 @@
+###############################################################################
+# ProteusLib Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+
+import pytest
+from watertap.property_models.tests.property_test_harness import \
+    (PropertyTestHarness, PropertyRegressionTest)
+from pyomo.environ import ConcreteModel, assert_optimal_termination, value
+from idaes.core import FlowsheetBlock
+import idaes.core.util.scaling as iscale
+from pyomo.util.check_units import assert_units_consistent
+from idaes.generic_models.properties.core.generic.generic_property import GenericParameterBlock
+from watertap.property_models.seawater_ion_generic import configuration
+from watertap.util.initialization import check_dof
+from idaes.core.util import get_solver
+
+solver = get_solver()
+# -----------------------------------------------------------------------------
+
+@pytest.mark.component
+def test_property_seawater_ions():
+    m = ConcreteModel()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.properties = GenericParameterBlock(default=configuration)
+    m.fs.stream = m.fs.properties.build_state_block([0], default={'defined_state': True})
+
+    # specify
+    # feed_flow_mass = 1
+    # feed_mass_frac = {'Na': 11122e-6,
+    #                   'Ca': 382e-6,
+    #                   'Mg': 1394e-6,
+    #                   'SO4': 2136e-6,
+    #                   'Cl': 20300e-6}
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'Na_+'].fix(0.008845)
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'Ca_2+'].fix(0.000174)
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'Mg_2+'].fix(0.001049)
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'SO4_2-'].fix(0.000407)
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'Cl_-'].fix(0.010479)
+    m.fs.stream[0].flow_mol_phase_comp['Liq', 'H2O'].fix(0.979046)
+    m.fs.stream[0].temperature.fix(273.15 + 25)
+    m.fs.stream[0].pressure.fix(101325)
+
+    # m.fs.stream[0].mass_frac_comp # TODO: mass_frac_comp and mass_frac_phase_comp need to be added to generic property model
+
```

Comment:
TODO comment here. Has this been addressed? Should remove this comment if it is no longer needed.

Reply:
It will need to be addressed in IDAES, and I has this here as a reminder for late. However, I noted it down and will remove.